### PR TITLE
Added dependency injection to the Application

### PR DESCRIPTION
**Pull request by vieck**
- Added Dagger to the app/build.gradle
 - Updated gradle build while i was in there
- Added Injection package
   - Created AppComponent to inject Activities and Fragments
  - Created AppModule to provide Singletons
- Injecting MachineService instead of getting it from the API

### CODE REVIEW

File: app/src/main/java/xyz/jhughes/laundry/AnalyticsApplication.java [hunk 1]
```diff
@@ -4,21 +4,33 @@
...
+ * <p>
  * It handles initializing the Analytics tracker.
  */
 public class AnalyticsApplication extends Application {
+    private AppComponent appComponent;
+
     @Override
     public void onCreate() {
         super.onCreate();
+        appComponent = DaggerAppComponent.builder()
```
Comment (vieck) [inline app/src/main/java/xyz/jhughes/laundry/AnalyticsApplication.java:23]: This is the starting point for all of Dagger

File: app/src/main/java/xyz/jhughes/laundry/apiclient/MachineAPI.java [hunk 2]
```diff
@@ -1,39 +1,15 @@
...
-    Call<Map<String,MachineList>> getAllMachines();
-
-    @GET("/v2/locations")
-    Call<List<Locations>> getLocations();
-
-    @GET("/v2-debug/location/{location}")
-    Call<ArrayList<Machine>> getMachineStatus_DEBUG(
-            @Path("location") String location
-    );
+public class MachineAPI {
```
Comment (vieck) [inline app/src/main/java/xyz/jhughes/laundry/apiclient/MachineAPI.java:13]: This file would have the retrofit request to the server. I can either do that in this PR or in a separate one but we will probably have to bring in rxjava if i am to do it now

Comment (vieck) [reply app/src/main/java/xyz/jhughes/laundry/apiclient/MachineAPI.java:13]: I added the methods for getting the Retrofit callback.

File: app/src/main/java/xyz/jhughes/laundry/AnalyticsApplication.java [hunk 3]
```diff
@@ -4,21 +4,33 @@
...
+import xyz.jhughes.laundry.injection.AppComponent;
+import xyz.jhughes.laundry.injection.AppModule;
+import xyz.jhughes.laundry.injection.DaggerAppComponent;
 
 /**
  * Base activity for the app.
- *
+ * <p>
  * It handles initializing the Analytics tracker.
  */
```
Comment (jpitlor) [inline app/src/main/java/xyz/jhughes/laundry/AnalyticsApplication.java:16]:
```suggestion
 * </p>
 */
```

File: app/src/main/java/xyz/jhughes/laundry/adapters/MachineAdapter.java [hunk 4]
```diff
@@ -17,25 +17,26 @@
 import android.widget.Toast;
 
 import java.util.ArrayList;
```
Comment (jpitlor) [inline app/src/main/java/xyz/jhughes/laundry/adapters/MachineAdapter.java:19]: not needed(?)

File: app/src/main/java/xyz/jhughes/laundry/apiclient/MachineAPI.java [hunk 5]
```diff
@@ -1,39 +1,45 @@
...
+    public static final String API_ROOT = "http://laundry-api.sigapp.club";
+    private final boolean DEBUG = BuildConfig.DEBUG;
+    private MachineService machineService;
+
+    public MachineAPI(MachineService machineService) {
+        this.machineService = machineService;
+    }
+
+    public Call<List<LocationResponse>> getLocations() {
+        if (DEBUG) {
```
Comment (jpitlor) [inline app/src/main/java/xyz/jhughes/laundry/apiclient/MachineAPI.java:23]: hmmmm I'm guessing there's a much cleaner way of doing this debug thing...i'll look into it in the next couple days

**Review by jpitlor — CHANGES_REQUESTED** [review]
i think i commented on everything, but ill look more at all the PRs in the next day or 2

File: app/src/main/java/xyz/jhughes/laundry/apiclient/MachineAPI.java (hunk 5 above)
Comment (vieck) [reply app/src/main/java/xyz/jhughes/laundry/apiclient/MachineAPI.java:23]: More than likely. I just cleaned it up the best i could

Comment (vieck) [reply app/src/main/java/xyz/jhughes/laundry/apiclient/MachineAPI.java:23]: @piticent123 We might be able to inject a debugging variable using dagger

**Review by jpitlor — APPROVED** [review]

**Pull request merged**